share/config.fish

```diff
@@ -259,7 +205,42 @@ end
 # This used to be in etc/config.fish - keep it here to keep the semantics
 #
 
+# Adapt construct_path from the macOS /usr/libexec/path_helper
+# executable for fish; see
+# https://opensource.apple.com/source/shell_cmds/shell_cmds-203/path_helper/path_helper.c.auto.html .
+function __fish_construct_path -d "construct a path variable for path_helper"
+    set -l result
+
+    for path_file in $argv[2] $argv[3]/*
+        if test -f $path_file
+            while read -la entry
+                if not contains $entry $result
+                    set result $result $entry
+                end
+            end <$path_file
+        end
+    end
+
+    for entry in $$argv[1]
+        if not contains $entry $result
+            set result $result $entry
+        end
+    end
+
+    for x in $result
+        echo $x
```

Comment:
Note: This entire for-loop could also be replaced with `set -xg $argv[1] $result` - then the outer `set` becomes unnecessary.

Reply:
Thanks, done!